File: sample/use_cocoapods/objectivec/GNAdGoogleRewardMediationAdapterSample/GNAdGoogleRewardMediationAdapterSample/RewardRegacyViewController.m
//
//  RewardRegacyViewController.m
//  GNAdGoogleRewardMediationAdapterSample
//

#import "RewardRegacyViewController.h"
#include "Util.h"
@import GoogleMobileAds;

@interface RewardRegacyViewController () <GADRewardBasedVideoAdDelegate>

@property (weak, nonatomic) IBOutlet UIButton *buttonLoad;
@property (weak, nonatomic) IBOutlet UIButton *buttonShow;

@end

@implementation RewardRegacyViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    _buttonShow.enabled = NO;

    [GADRewardBasedVideoAd sharedInstance].delegate = self;
}

- (IBAction)downButtonLoad:(id)sender {
    _buttonLoad.enabled = NO;
    _buttonShow.enabled = NO;

    DFPRequest *request = [DFPRequest request];
    //request.testDevices = @[[Util admobDeviceID]];
    [[GADRewardBasedVideoAd sharedInstance] loadRequest:request withAdUnitID:_unitId];
}

- (IBAction)downButtonShow:(id)sender {
    if ([[GADRewardBasedVideoAd sharedInstance] isReady]) {
        [[GADRewardBasedVideoAd sharedInstance] presentFromRootViewController:self];
    }
}


#pragma mark: GADRewardBasedVideoAdDelegate

- (void)rewardBasedVideoAd:(GADRewardBasedVideoAd *)rewardBasedVideoAd didRewardUserWithReward:(GADAdReward *)reward {
    
    NSString *rewardMessage =
    [NSString stringWithFormat:@"Reward received with currency %@ , amount %lf", reward.type, [reward.amount doubleValue]];
    NSLog(@"rewardBasedVideoAd:didRewardUserWithReward %@", rewardMessage);
}

- (void)rewardBasedVideoAd:(GADRewardBasedVideoAd *)rewardBasedVideoAd didFailToLoadWithError:(NSError *)error {
    NSLog(@"rewardBasedVideoAd:didFailToLoadWithError error = %@", [error localizedDescription]);
    _buttonLoad.enabled = YES;
    _buttonShow.enabled = NO;
}

- (void)rewardBasedVideoAdDidReceiveAd:(GADRewardBasedVideoAd *)rewardBasedVideoAd {
    NSLog(@"rewardBasedVideoAdDidReceiveAd");
    _buttonLoad.enabled = YES;
    _buttonShow.enabled = YES;
}

/// Tells the delegate that the reward based video ad opened.
- (void)rewardBasedVideoAdDidOpen:(GADRewardBasedVideoAd *)rewardBasedVideoAd {
    NSLog(@"rewardBasedVideoAdDidOpen");
}

/// Tells the delegate that the reward based video ad started playing.
- (void)rewardBasedVideoAdDidStartPlaying:(GADRewardBasedVideoAd *)rewardBasedVideoAd {
    NSLog(@"rewardBasedVideoAdDidStartPlaying");
}

/// Tells the delegate that the reward based video ad completed playing.
- (void)rewardBasedVideoAdDidCompletePlaying:(GADRewardBasedVideoAd *)rewardBasedVideoAd {
    NSLog(@"rewardBasedVideoAdDidCompletePlaying");
}

/// Tells the delegate that the reward based video ad closed.
- (void)rewardBasedVideoAdDidClose:(GADRewardBasedVideoAd *)rewardBasedVideoAd {
    NSLog(@"rewardBasedVideoAdDidClose");
    _buttonShow.enabled = NO;
}

/// Tells the delegate that the reward based video ad will leave the application.
- (void)rewardBasedVideoAdWillLeaveApplication:(GADRewardBasedVideoAd *)rewardBasedVideoAd {
    NSLog(@"rewardBasedVideoAdWillLeaveApplication");
}

@end

File: sample/use_cocoapods/swift/GNAdGoogleRewardAdapterSample/GNAdGoogleRewardAdapterSample/RewardRegacyViewController.swift
//
//  RewardRegacyViewController.swift
//  GNAdGoogleRewardAdapterSample
//

import UIKit
import GoogleMobileAds

class RewardRegacyViewController: UIViewController {

    var unitId: String!

    @IBOutlet weak var buttonLoad: UIButton!
    @IBOutlet weak var buttonShow: UIButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        buttonShow.isEnabled = false

        GADRewardBasedVideoAd.sharedInstance().delegate = self
    }

    @IBAction func downButtonLoad(_ sender: Any) {
        buttonLoad.isEnabled = false
        buttonShow.isEnabled = false

        let request = DFPRequest()
        //request.testDevices = [ Util.admobDeviceID() ]
        GADRewardBasedVideoAd.sharedInstance().load(request, withAdUnitID: unitId)
    }
    
    @IBAction func downButtonShow(_ sender: Any) {
        if (GADRewardBasedVideoAd.sharedInstance().isReady) {
            GADRewardBasedVideoAd.sharedInstance().present(fromRootViewController: self)
        }
    }

}
// MARK: GADRewardBasedVideoAdDelegate implementation
extension RewardRegacyViewController : GADRewardBasedVideoAdDelegate {
    
    func rewardBasedVideoAd(_ rewardBasedVideoAd: GADRewardBasedVideoAd,
                            didRewardUserWith reward: GADAdReward) {
        print("rewardBasedVideoAd:didRewardUserWith \(reward.type), amount \(reward.amount).")
    }

    func rewardBasedVideoAd(_ rewardBasedVideoAd: GADRewardBasedVideoAd,
                            didFailToLoadWithError error: Error) {
        print("rewardBasedVideoAd:didFailToLoadWithError error =  \(error.localizedDescription)")
        buttonLoad.isEnabled = true
        buttonShow.isEnabled = false
    }

    func rewardBasedVideoAdDidReceive(_ rewardBasedVideoAd: GADRewardBasedVideoAd) {
        print("rewardBasedVideoAdDidReceive")
        buttonLoad.isEnabled = true
        buttonShow.isEnabled = true
    }

    func rewardBasedVideoAdDidOpen(_ rewardBasedVideoAd: GADRewardBasedVideoAd) {
        print("rewardBasedVideoAdDidOpen")
    }

    func rewardBasedVideoAdDidStartPlaying(_ rewardBasedVideoAd: GADRewardBasedVideoAd) {
        print("rewardBasedVideoAdDidStartPlaying : " + (rewardBasedVideoAd.adNetworkClassName?.description)!)
    }

    func rewardBasedVideoAdDidCompletePlaying(_ rewardBasedVideoAd: GADRewardBasedVideoAd) {
        print("rewardBasedVideoAdDidCompletePlaying")
    }

    func rewardBasedVideoAdDidClose(_ rewardBasedVideoAd: GADRewardBasedVideoAd) {
        print("rewardBasedVideoAdDidClose")
    }

    func rewardBasedVideoAdWillLeaveApplication(_ rewardBasedVideoAd: GADRewardBasedVideoAd) {
        print("rewardBasedVideoAdWillLeaveApplication")
    }

}
